src/js/jsx/shared/Dialog.jsx

```diff
@@ -124,11 +132,10 @@ define(function (require, exports, module) {
                 id = this.props.id;
 
             if (this.state.open) {
+                this._target = null;
                 flux.actions.dialog.closeDialog(id);
             } else if (!this.props.disabled) {
-                this.setState({
-                    target: event.target
-                });
+                this._target = event.target;
```

Comment:
This *seems* safe to me, but maybe I'm missing something.  From [the docs](http://facebook.github.io/react/docs/component-api.html):
"The second (optional) parameter is a callback function that will be executed once setState is completed and the component is re-rendered."